packages/squarev2-integration/src/squarev2-payment-strategy.ts

```diff
@@ -29,8 +30,17 @@ export default class SquareV2PaymentStrategy implements PaymentStrategy {
         const { methodId, squarev2 } = options;
         const {
             config: { testMode },
-            initializationData: { applicationId, locationId },
-        } = this._paymentIntegrationService.getState().getPaymentMethodOrThrow(methodId);
+            initializationData,
+        } = this._paymentIntegrationService
+            .getState()
+            .getPaymentMethodOrThrow<SquarePaymentMethodInitializationData>(methodId);
+        const { applicationId, locationId } = initializationData || {};
+
+        if (!applicationId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "applicationId" argument is not provided.',
+            );
```

Comment:
It might be better to use another error here, because it is not an issue with provided options from checkout js or other plcase to checkout sdk. Consequently, it is an issue with payment method configuration on CP or so.

What do you think?

```suggestion
           throw new PaymentMethodInvalidError();
```

Reply:
thanks, changed